src/display/node_utils.js

```diff
@@ -28,37 +28,81 @@ if (typeof PDFJSDev !== "undefined" && PDFJSDev.test("MOZCENTRAL")) {
   );
 }
 
+if (isNodeJS && !globalThis.__pdfjsPackages__) {
+  let fs, http, https, url, canvas, path2d_polyfill;
+
+  if (typeof PDFJSDev !== "undefined" && PDFJSDev.test("LIB")) {
```

Comment:
I think we can avoid duplication in this block by defining `const loader = (typeof PDFJSDev !== "undefined" && PDFJSDev.test("LIB")) ? __non_webpack_require__ : __non_webpack_import__;` and then using e.g. `const fs = loader("fs");` so we only have to declare the modules once.

Reply:
Unfortunately I don't *think* that'll work, since the `import` statements use `await` and the "top level await"-functionality is only available in proper modules.

Something that I forgot to mention is that it *should* be possible to simplify this code further, by removing the `LIB`-branches altogether, and directly test the source-code in Node.js for the GitHub-actions tests.
However, given the size of this PR I didn't want to attempt that here as well since I was able to "hack around" it for now.

Reply:
By the way, thanks for taking the time to review all of this!

Reply:
> Unfortunately I don't think that'll work, since the import statements use await and the "top level await"-functionality is only available in proper modules.

Ah, you're completely right. I somehow missed the `await` usage here.

Reply:
Note that these, and other Node.js specific, imports were simplified in PR #17114.